### PR TITLE
Fixed custom product subscription related tests

### LINKED ISSUE #3857: Fix custom product subscription related tests
List of affected tests (at least these, there can be even more affected):
```
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_install_package
tests.foreman.cli.test_host.KatelloAgentTestCase.test_negative_unregister_and_pull_content
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_apply_errata
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_get_errata_info
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_install_package_group
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_remove_package
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_remove_package_group
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_upgrade_package
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_upgrade_packages_all
tests.foreman.api.test_errata.ErrataTestCase.test_positive_filter_by_cve
tests.foreman.api.test_errata.ErrataTestCase.test_positive_filter_by_envs
tests.foreman.api.test_errata.ErrataTestCase.test_positive_get_applicable_for_host
tests.foreman.api.test_errata.ErrataTestCase.test_positive_get_count_for_host
tests.foreman.api.test_errata.ErrataTestCase.test_positive_get_diff_for_cv_envs
tests.foreman.api.test_errata.ErrataTestCase.test_positive_install_in_hc
tests.foreman.api.test_errata.ErrataTestCase.test_positive_install_in_host
tests.foreman.api.test_errata.ErrataTestCase.test_positive_list
tests.foreman.api.test_errata.ErrataTestCase.test_positive_list_updated
tests.foreman.api.test_errata.ErrataTestCase.test_positive_sort_by_issued_date
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_view_cve
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_search_autocomplete
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_view_products_and_repos
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_list_permission
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_list
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_chost_library
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_apply_for_all_hosts
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_apply_for_host
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_chost_previous_env
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_show_count_on_chost_details_page
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_show_count_on_chost_page
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_install_errata
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_install_package
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_install_package_group
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_remove_package
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_remove_package_group
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_upgrade_package
```

All of them are failing with the same error message:
```python
>               if int(subscription['quantity']) == 0:
E               ValueError: invalid literal for int() with base 10: 'Unlimited'
```

### PR DESCRIPTION
**Pull request by abalakh**
Closes #3857
The root cause is custom products subscriptions in 6.3 may contain
string value 'Unlimited' for its 'quantity' field, and robottelo
was expecting only string containing integer and was converting it
with `int()` func, which leads to failure.

List of affected tests (not complete):
```
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_install_package
tests.foreman.cli.test_host.KatelloAgentTestCase.test_negative_unregister_and_pull_content
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_apply_errata
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_get_errata_info
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_install_package_group
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_remove_package
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_remove_package_group
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_upgrade_package
tests.foreman.cli.test_host.KatelloAgentTestCase.test_positive_upgrade_packages_all
tests.foreman.api.test_errata.ErrataTestCase.test_positive_filter_by_cve
tests.foreman.api.test_errata.ErrataTestCase.test_positive_filter_by_envs
tests.foreman.api.test_errata.ErrataTestCase.test_positive_get_applicable_for_host
tests.foreman.api.test_errata.ErrataTestCase.test_positive_get_count_for_host
tests.foreman.api.test_errata.ErrataTestCase.test_positive_get_diff_for_cv_envs
tests.foreman.api.test_errata.ErrataTestCase.test_positive_install_in_hc
tests.foreman.api.test_errata.ErrataTestCase.test_positive_install_in_host
tests.foreman.api.test_errata.ErrataTestCase.test_positive_list
tests.foreman.api.test_errata.ErrataTestCase.test_positive_list_updated
tests.foreman.api.test_errata.ErrataTestCase.test_positive_sort_by_issued_date
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_view_cve
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_search_autocomplete
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_view_products_and_repos
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_list_permission
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_list
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_chost_library
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_apply_for_all_hosts
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_apply_for_host
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_chost_previous_env
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_show_count_on_chost_details_page
tests.foreman.ui.test_errata.ErrataTestCase.test_positive_show_count_on_chost_page
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_install_errata
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_install_package
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_install_package_group
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_remove_package
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_remove_package_group
tests.foreman.ui.test_contenthost.ContentHostTestCase.test_positive_upgrade_package
```

Ran 2 random tests from the list:
```python
/home/andrii/env/bin/python /home/andrii/workspace/pycharm-2016.1.3/helpers/pydev/pydevd.py --multiproc --qt-support --client 127.0.0.1 --port 33839 --file /home/andrii/workspace/pycharm-2016.1.3/helpers/pycharm/pytestrunner.py -p pytest_teamcity /home/andrii/workspace/robottelo/tests/foreman/ui/test_errata.py "-k ErrataTestCase and test_positive_list"
Testing started at 4:42 PM ...
pydev debugger: process 7666 is connecting

Connected to pydev debugger (build 145.1504)
============================= test session starts ==============================
platform linux2 -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
collected 17 items

..


====== 15 tests deselected by '-k ErrataTestCase and test_positive_list' =======
================== 2 passed, 15 deselected in 1619.40 seconds ==================

Process finished with exit code 0
```

### CODE REVIEW

File: tests/foreman/ui/test_errata.py [hunk 1]
```diff
@@ -156,13 +156,13 @@ def test_positive_list(self):
         with Session(self.browser) as session:
             session.nav.go_to_errata()
             self.errata.show_only_applicable(False)
-            self.assertIsNone(self.errata.search(CUSTOM_REPO_ERRATA_ID))
-            self.assertIsNotNone(self.errata.search(FAKE_1_ERRATA_ID))
+            self.assertIsNone(self.errata.search(FAKE_3_ERRATA_ID))
+            self.assertIsNotNone(self.errata.search(CUSTOM_REPO_ERRATA_ID))
```
Comment (abalakh) [inline tests/foreman/ui/test_errata.py]: This change is basically bonus. When we changed from sat6 tools repo to sat6.2 tools repo errata ids became outdated so the test was failing as it couldn't find mentioned errata ids.

File: tests/foreman/ui/test_errata.py [hunk 2]
```diff
@@ -156,13 +156,13 @@ def test_positive_list(self):
         with Session(self.browser) as session:
             session.nav.go_to_errata()
             self.errata.show_only_applicable(False)
-            self.assertIsNone(self.errata.search(CUSTOM_REPO_ERRATA_ID))
-            self.assertIsNotNone(self.errata.search(FAKE_1_ERRATA_ID))
+            self.assertIsNone(self.errata.search(FAKE_3_ERRATA_ID))
```
Comment (omaciel) [inline tests/foreman/ui/test_errata.py]: I see that the logic has been inverted here (not just the order but the logic)... why the change?

File: tests/foreman/ui/test_errata.py [hunk 3]
```diff
@@ -156,13 +156,13 @@ def test_positive_list(self):
...
+            self.assertIsNotNone(self.errata.search(CUSTOM_REPO_ERRATA_ID))
             session.nav.go_to_select_org(org.name)
             session.nav.go_to_errata()
             self.errata.show_only_applicable(False)
-            self.assertIsNone(self.errata.search(FAKE_1_ERRATA_ID))
-            self.assertIsNotNone(self.errata.search(CUSTOM_REPO_ERRATA_ID))
+            self.assertIsNone(self.errata.search(CUSTOM_REPO_ERRATA_ID))
```
Comment (omaciel) [inline tests/foreman/ui/test_errata.py]: I see that the logic has been inverted here (not just the order but the logic)... why the change?

File: tests/foreman/ui/test_errata.py (hunk 2 above)
Comment (abalakh) [inline tests/foreman/ui/test_errata.py]: I tried to answer it in my comment [above](https://github.com/SatelliteQE/robottelo/pull/3858#discussion_r80494703) (1 min before yours comment so i suspect github didn't show you it in time).
Just saw the errata ids are outdated because of different repos we use now, carefully checked what's actually available in the first and the second organization and updated the test with corresponding changes.

**Comment by oshtaier** [comment]
ACK
